Userland/Libraries/LibWeb/HTML/FormControlInfrastructure.cpp

```diff
@@ -40,4 +46,140 @@ WebIDL::ExceptionOr<Entry> create_entry(JS::Realm& realm, DeprecatedString const
+            // FIXME: 1. If the field element has a name attribute specified and its value is not the empty string, let name be that value followed by a single U+002E FULL STOP character (.). Otherwise, let name be the empty string.
+            // FIXME: 2. Let namex be the string consisting of the concatenation of name and a single U0078 LATIN SMALL LETTER X character (x).
+            // FIXME: 3. Let namey be the string consisting of the concatenation of name and a single U+0079 LATIN SMALL LETTER Y character (y).
+            // FIXME: 4. The field element is submitter, and before this algorithm was invoked the user indicated a coordinate. Let x be the x-component of the coordinate selected by the user, and let y be the y-component of the coordinate selected by the user.
+            // FIXME: 5. Create an entry with namex and x, and append it to entry list.
+            // FIXME: 6. Create an entry with namey and y, and append it to entry list.
+            // 7. Continue.
+            continue;
+        }
+
+        // FIXME: 3. If the field is a form-associated custom element, then perform the entry construction algorithm given field and entry list, then continue.
+
+        // 4. If either the field element does not have a name attribute specified, or its name attribute's value is the empty string, then continue.
+        if (control->name().is_empty())
+            continue;
+
+        // 5. Let name be the value of the field element's name attribute.
+        auto name = TRY_OR_THROW_OOM(realm.vm(), Infra::convert_to_scalar_value_string(control->name()));
+
+        auto form_data_entries = entry_list.contains(name) && entry_list.get(name).has_value()
+            ? entry_list.get(name).value()
+            : Vector<Variant<JS::Handle<FileAPI::File>, DeprecatedString>> {};
+
+        // 6. If the field element is a select element, then for each option element in the select element's list of options whose selectedness is true and that is not disabled, create an entry with name and the value of the option element, and append it to entry list.
+        if (auto* select_element = dynamic_cast<HTML::HTMLSelectElement*>(control.ptr())) {
+            for (auto const& option_element : select_element->list_of_options()) {
+                if (option_element->selected() && !option_element->disabled())
+                    TRY_OR_THROW_OOM(realm.vm(), form_data_entries.try_append(option_element->value()));
+            }
+            TRY_OR_THROW_OOM(realm.vm(), entry_list.try_set(name, form_data_entries));
+        }
+        // 7. Otherwise, if the field element is an input element whose type attribute is in the Checkbox state or the Radio Button state, then:
+        else if (auto* input_element = dynamic_cast<HTML::HTMLInputElement*>(control.ptr()); input_element && (input_element->type() == "checkbox" || input_element->type() == "radio") && input_element->checked()) {
```

Comment:
The spec does not mention `checked` here, but I agree it seems like it should.
Is this a (known) spec issue?